specs/bls_signature.md

```diff
@@ -33,6 +31,18 @@ The BLS12-381 curve parameters are defined [here](https://z.cash/blog/new-snark-
 
 We represent points in the groups G1 and G2 following [zkcrypto/pairing](https://github.com/zkcrypto/pairing/tree/master/src/bls12_381). We denote by `q` the field modulus and by `i` the imaginary unit.
 
+## Ciphersuite
+
+We use the following Ciphersuite `BLS_SIG_BLS12381G2-SHA256-SSWU-RO_POP_`.
```

Comment:
Should this ciphersuite only be used for the proof of possession (`deposit_data.signature`)?

Reply:
Should the ciphersuite be `BLS_SIG_BLS12381G2-SHA256-SSWU-RO-_POP_` (notice the extra `-` after `RO-`)?

Reply:
I don't think so, the IETF spec has a `hash_to_point` definition for messages of this ciphersuite which is distinct from `hash_pubkey_to_point`. My understanding is that if you use the [PoP defined by the IETF spec](https://tools.ietf.org/html/draft-irtf-cfrg-bls-signature-00#section-4.2.3), you should use `BLS_SIG_BLS12381G2-SHA256-SSWU-RO_POP_` or `BLS_SIG_BLS12381G2-SHA256-SSWU-RO-_NUL_` if their PoP is not used.

Considering that we do not use the exact PoP defined by the IETF spec, I think we should not claim to use the PoP ciphersuite anywhere.

```suggestion
We use the following Ciphersuite `BLS_SIG_BLS12381G2-SHA256-SSWU-RO-_NUL_`.
```

Reply:
I believe we should just use one ciphersuite for both PoP and regular signature validation.
The extra `-` after `RO-` is deliberate as that's how the examples were given in the IETF spec.

I agree with Carl, if we don't use the specified PoP we should not use that in our ciphersuite tag. However, I then ask why do we not use that standardized PoP?

Reply:
> The extra - after RO- is deliberate as that's how the examples were given in the IETF spec.

Right :) The extra `-` were missing and added them in.

> why do we not use that standardized PoP?

What is the difference between the standardized PoP and the currently specified PoP?

Reply:
Good question,
The standardize PoP signs the message:
`msg = bytes(public_key)`

Where as we sign the message:
`msg =  signing_root(deposit.data)`
Which contains `withdrawal_credentials` and `amount` along with the `public_key`.

Reply:
> The standardize PoP signs the message:
> `msg = bytes(public_key)`

Note that there is a specific domain separation tags just for signing pubkeys. (A third tag not mentioned here yet.)

Reply:
You're right sorry, I didn't notice the tag was slightly different between regular signatures and PoP.
PoP: `BLS_POP_BLS12381G2-SHA256-SSWU-RO-_POP_`
Regular Signature: `BLS_SIG_BLS12381G2-SHA256-SSWU-RO-_POP_`

With the option of changing the last `_POP_` to `_NUL_` if required.